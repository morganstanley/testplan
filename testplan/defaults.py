"""All default values that will be shared between config objects go here."""

import os

from testplan.report.testing.styles import StyleArg

TESTPLAN_TIMEOUT = 14400  # 4h

SUMMARY_NUM_PASSING = 5
SUMMARY_NUM_FAILING = 5
SUMMARY_KEY_COMB_LIMIT = 10  # Number of failed key combinations to summary.

# Make sure these values match the defaults in the parser.py,
# otherwise we may end up with inconsistent behaviour re. defaults
# between cmdline and programmatic calls.
PDF_STYLE = StyleArg.SUMMARY.value
STDOUT_STYLE = StyleArg.EXTENDED_SUMMARY.value

REPORT_DIR = os.getcwd()
XML_DIR = os.path.join(REPORT_DIR, "xml")
PDF_PATH = os.path.join(REPORT_DIR, "report.pdf")
JSON_PATH = os.path.join(REPORT_DIR, "report.json")
ATTACHMENTS = "_attachments"
RESOURCE_DATA = "_resource"
RESOURCE_META_FILE_NAME = "metadata.json"
ATTACHMENTS_DIR = os.path.join(REPORT_DIR, ATTACHMENTS)
RESOURCE_DATA_DIR = os.path.join(REPORT_DIR, RESOURCE_DATA)

WEB_SERVER_HOSTNAME = "0.0.0.0"
WEB_SERVER_PORT = 0
WEB_SERVER_TIMEOUT = 10

# Default to using 4 threads for interactive pool.
INTERACTIVE_POOL_SIZE = 4

# Name of multitest/testsuite/testcase (usually used for display) cannot be
# too long, or the UI will not be pleasant when they end up with long names
MAX_TEST_NAME_LENGTH = 255

# Default to using 30 min for auto-parts task.
AUTO_PART_RUNTIME_MAX = 30 * 60
# Each part shall be at least 8 min
AUTO_PART_RUNTIME_MIN = 8 * 60
# Expect to finish within 3 times of the longest start+stop time
START_STOP_FACTOR = 3

# Default to using 30min for calculating pool size
PLAN_RUNTIME_TARGET = 1800  # 30 min
