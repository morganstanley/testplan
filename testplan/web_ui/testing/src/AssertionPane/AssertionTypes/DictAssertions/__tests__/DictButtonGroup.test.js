import React from "react";
import { render } from "@testing-library/react";
import { StyleSheetTestUtils } from "aphrodite";

import { SORT_TYPES, FILTER_OPTIONS } from "./../../../../Common/defaults";
import DictButtonGroup from "../DictButtonGroup";

function defaultProps() {
  return {
    uid: "sort-test-uid",
    defaultSortType: SORT_TYPES.BY_STATUS,
    sortTypeList: [
      SORT_TYPES.NONE,
      SORT_TYPES.ALPHABETICAL,
      SORT_TYPES.REVERSE_ALPHABETICAL,
      SORT_TYPES.BY_STATUS,
    ],
    filterOptionList: [
      FILTER_OPTIONS.FAILURES_ONLY,
      FILTER_OPTIONS.EXCLUDE_IGNORABLE,
    ],
    flattenedDict: [
      [0, "foo", "Passed", ["int", "1"], ["int", "1"]],
      [0, "bar", "Failed", ["int", "2"], ["int", "5"]],
      [0, "extra-key", "Failed", [null, "ABSENT"], ["int", "10"]],
    ],
    setRowData: jest.fn(),
  };
}

describe("DictLogAssertion", () => {
  let props;
  let component;

  beforeEach(() => {
    // Stop Aphrodite from injecting styles, this crashes the tests.
    StyleSheetTestUtils.suppressStyleInjection();
    props = {};
    component = undefined;
  });

  it("renders DictButtonGroup component", () => {
    props = defaultProps();
    component = render(<DictButtonGroup {...props} />);
    expect(component.asFragment()).toMatchSnapshot();
  });
});
