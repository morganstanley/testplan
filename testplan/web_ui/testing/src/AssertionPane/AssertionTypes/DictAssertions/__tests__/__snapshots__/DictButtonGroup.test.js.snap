// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DictLogAssertion renders DictButtonGroup component 1`] = `
<DocumentFragment>
  <div
    class="btn-group"
    role="group"
    style="padding-bottom: .5rem;"
  >
    <button
      class="btn btn-outline-secondary btn-sm"
      type="button"
    >
      Original order
    </button>
    <button
      class="btn btn-outline-secondary btn-sm"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-sort-amount-down fa-w-16 fa-sm "
        data-icon="sort-amount-down"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M187.298 395.314l-79.984 80.002c-6.248 6.247-16.383 6.245-22.627 0L4.705 395.314C-5.365 385.244 1.807 368 16.019 368H64V48c0-8.837 7.163-16 16-16h32c8.837 0 16 7.163 16 16v320h47.984c14.241 0 21.363 17.264 11.314 27.314zM240 96h256c8.837 0 16-7.163 16-16V48c0-8.837-7.163-16-16-16H240c-8.837 0-16 7.163-16 16v32c0 8.837 7.163 16 16 16zm-16 112v-32c0-8.837 7.163-16 16-16h192c8.837 0 16 7.163 16 16v32c0 8.837-7.163 16-16 16H240c-8.837 0-16-7.163-16-16zm0 256v-32c0-8.837 7.163-16 16-16h64c8.837 0 16 7.163 16 16v32c0 8.837-7.163 16-16 16h-64c-8.837 0-16-7.163-16-16zm0-128v-32c0-8.837 7.163-16 16-16h128c8.837 0 16 7.163 16 16v32c0 8.837-7.163 16-16 16H240c-8.837 0-16-7.163-16-16z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="btn btn-outline-secondary btn-sm"
      type="button"
    >
      <svg
        aria-hidden="true"
        class="svg-inline--fa fa-sort-amount-up fa-w-16 fa-sm "
        data-icon="sort-amount-up"
        data-prefix="fas"
        role="img"
        viewBox="0 0 512 512"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M4.702 116.686l79.984-80.002c6.248-6.247 16.383-6.245 22.627 0l79.981 80.002c10.07 10.07 2.899 27.314-11.314 27.314H128v320c0 8.837-7.163 16-16 16H80c-8.837 0-16-7.163-16-16V144H16.016c-14.241 0-21.363-17.264-11.314-27.314zM240 96h256c8.837 0 16-7.163 16-16V48c0-8.837-7.163-16-16-16H240c-8.837 0-16 7.163-16 16v32c0 8.837 7.163 16 16 16zm-16 112v-32c0-8.837 7.163-16 16-16h192c8.837 0 16 7.163 16 16v32c0 8.837-7.163 16-16 16H240c-8.837 0-16-7.163-16-16zm0 256v-32c0-8.837 7.163-16 16-16h64c8.837 0 16 7.163 16 16v32c0 8.837-7.163 16-16 16h-64c-8.837 0-16-7.163-16-16zm0-128v-32c0-8.837 7.163-16 16-16h128c8.837 0 16 7.163 16 16v32c0 8.837-7.163 16-16 16H240c-8.837 0-16-7.163-16-16z"
          fill="currentColor"
        />
      </svg>
    </button>
    <button
      class="btn btn-outline-secondary btn-sm active"
      type="button"
    >
      Status
    </button>
    <button
      class="btn btn-outline-secondary btn-sm"
      type="button"
    >
      Failures only
    </button>
    <button
      class="btn btn-outline-secondary btn-sm"
      type="button"
    >
      Hide ignored items
    </button>
    <button
      class="btn btn-outline-secondary btn-sm"
      type="button"
    >
      Copy
    </button>
  </div>
</DocumentFragment>
`;
